### PR TITLE
SONARJAVA-3665 Add support of Text Blocks in S2973 (Escaped unicode characters)

### PR DESCRIPTION
**Pull request by margarita-nedzelska-sonarsource**
Waits for https://github.com/SonarSource/sonar-java/pull/3399 to be merged

### CODE REVIEW

File: java-checks/src/test/java/org/sonar/java/checks/EscapedUnicodeCharactersCheckTest.java [hunk 1]
```diff
@@ -34,4 +35,12 @@ void test() {
       .verifyIssues();
   }
 
+  @Test
+  void testTextBlocks() {
```
Comment (dorian-burihabwa-sonarsource) [inline java-checks/src/test/java/org/sonar/java/checks/EscapedUnicodeCharactersCheckTest.java:39]: We should rename this method in snake_case for consistency.

File: java-checks/src/main/java/org/sonar/java/checks/EscapedUnicodeCharactersCheck.java [hunk 2]
```diff
@@ -54,6 +54,9 @@ public void visitNode(Tree node) {
       return;
     }
     String value = LiteralUtils.trimQuotes(((LiteralTree) node).value());
+    if (node.is(Kind.TEXT_BLOCK)) {
+      value = value.replaceAll("(\r?\n|\r)\\s*+", "");
```
Comment (dorian-burihabwa-sonarsource) [inline java-checks/src/main/java/org/sonar/java/checks/EscapedUnicodeCharactersCheck.java:58]: Is the `+`  operator necessary here? The '*' just before already covers the same case. Or is the `+` supposed o apply to the entire expression?

File: java-checks-test-sources/src/main/files/non-compiling/checks/EscapedUnicodeCharactersTextBlocksCheck.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
+package checks;
+
+class EscapedUnicodeCharactersCheck {
+  void method() {
+    String prefix = """
+      n\uuuu00E9e"""; // Noncompliant@-1 [[sc=21;ec=20]] {{Remove this Unicode escape sequence and use the character instead.}}
```
Comment (dorian-burihabwa-sonarsource) [inline java-checks-test-sources/src/main/files/non-compiling/checks/EscapedUnicodeCharactersTextBlocksCheck.java:6]: We should probably include examples where multiple line breaks follow one another.
```java
    String prefix = """


      n\uuuu00E9e

""";
```

**Review by dorian-burihabwa-sonarsource — CHANGES_REQUESTED** [review]

File: java-checks/src/main/java/org/sonar/java/checks/EscapedUnicodeCharactersCheck.java (hunk 2 above)
Comment (margarita-nedzelska-sonarsource) [reply java-checks/src/main/java/org/sonar/java/checks/EscapedUnicodeCharactersCheck.java:58]: Yes, possessive quantifier is odd in this context, copy/paste issue

**Review by dorian-burihabwa-sonarsource — APPROVED** [review]
LGTM :+1:  Merging now

**Pull request merged**